diff --git a/rsky-relay/src/config.rs b/rsky-relay/src/config.rs
--- a/rsky-relay/src/config.rs
+++ b/rsky-relay/src/config.rs
@@ -30,7 +30,10 @@
 pub const WORKERS_PUBLISHERS: usize = 4;
 
 // server
-pub const PORT: u16 = if cfg!(feature = "labeler") { 9001 } else { 9000 };
+pub static PORT: LazyLock<u16> = LazyLock::new(|| {
+    env::var("RELAY_PORT").ok().and_then(|s| s.parse().ok())
+        .unwrap_or(if cfg!(feature = "labeler") { 9001 } else { 9000 })
+});
 pub const HOSTS_RELAY: &str = "relay1.us-west.bsky.network";
 pub const HOSTS_RELAYS_DEFAULT: &[&str] =
     &["relay1.us-west.bsky.network", "relay1.us-east.bsky.network", "bsky.network"];
diff --git a/rsky-relay/src/server/server.rs b/rsky-relay/src/server/server.rs
--- a/rsky-relay/src/server/server.rs
+++ b/rsky-relay/src/server/server.rs
@@ -205,7 +205,7 @@
             None
         };
 
-        let listener = TcpListener::bind(format!("0.0.0.0:{PORT}"))?;
+        let listener = TcpListener::bind(format!("0.0.0.0:{}", *PORT))?;
         listener.set_nonblocking(true)?;
         let base_url = Url::parse("http://example.com")?;
         let now = Instant::now();
